.travis.yml

```diff
@@ -1,4 +1,4 @@
-sudo: false
+sudo: required
```

Comment:
The CNAME record solution let you deal with collisions. You'll get that from the AvahiEntryGroupState that is passed back to callbacks like http://avahi.org/download/doxygen/publish_8h.html#afbd8ce431f0416537fc7b6101e0f9a1c

That being said I don't have anything against using DBus. I've only seen the argument about it being to heavy in the context of small embedded platforms, but that's not our case.